packages/@aws-cdk/custom-resources/lib/provider-framework/runtime/framework.ts

```diff
@@ -112,6 +120,23 @@ async function invokeUserFunction<A extends { ResponseURL: '...' }>(functionArnE
 
   const jsonPayload = parseJsonPayload(resp.Payload);
   if (resp.FunctionError) {
+    if (resp.FunctionError.includes('Lambda is initializing your function')) {
```

Comment:
Do we know that this part of the error message isn't used in other errors? It wouldn't be a great experience if this caught other errors as well and then the customer would have to wait ~10 minutes to find out that their deployment failed. I think it would be better, on an error, to check the function state per [Lambda Function States](https://docs.aws.amazon.com/lambda/latest/dg/functions-states.html). If the function is inactive or pending, we can retry and/or look at `StateReasonCode` to further narrow the retry criteria. If it's active or failed, we return the terminal result.